src/dbus_fast/aio/message_reader.py

```diff
@@ -0,0 +1,35 @@
+import logging
+import traceback
+from typing import Callable, Optional
+
+from .._private.unmarshaller import Unmarshaller
+from ..message import Message
+
+
+def build_message_reader(
+    unmarshaller: Unmarshaller,
+    process: Callable[[Message], None],
+    finalize: Callable[[Optional[Exception]], None],
+) -> None:
+    """Build a callable that reads messages from the unmarshaller and passes them to the process function."""
+    unmarshall = unmarshaller.unmarshall
+    reset = unmarshaller.reset
+
+    def _message_reader() -> None:
```

Comment:
The closure is faster than the bound object and we speed it up even more by pre-unpacking unmarshall and reset since these can get called 20000 times per minute